### PR TITLE
[Draft] Fix SITL offboard attitude for VTOL_TYPE_RESERVED2

### LINKED ISSUE #13092: [VTOL] Offboard attitude control not working
**Describe the bug**
Based on [discussion 13159](https://discuss.px4.io/t/vtol-hil-with-offboard-attitude-setpoints-not-working/13159).

VTOLs don't follow the offboard attitude setpoints, instead they keep the last attitude setpoint from before switching to offboard mode.

For more details (and other issues encountered), see the above-mentioned post on discuss.px4.io.

**To Reproduce**
Run the [MAVSDK offboard example](https://mavsdk.mavlink.io/develop/en/examples/offboard_velocity.html).

I have tested it on SITL and HIL on the standard VTOL model (with a slightly modified version of the MAVSDK example), in both simulations there are issues. Haven't dared to try it on a real drone, pretty sure it also doesn't work.

**Expected behavior**
Drone should follow the attitude setpoints that are passed via MAVLink.

**Log Files and Screenshots**
See [post](https://discuss.px4.io/t/vtol-hil-with-offboard-attitude-setpoints-not-working/13159) on Discuss.

**Drone (please complete the following information):**
- Standard VTOL (Gazebo)

**Additional context**

Main findings so far:
- The problem can be solved for SITL by settting the correct ORB ID (`mc_virtual_attitude_setpoint`) for `_att_sp_pub` in `mavlink_receiver.h` (need to set it so that it still works for non-VTOLs)
- The SITL solution doesn't work for HIL (tested with Pixhawk 4 as hardware). (not tested on real vehicles)
- Additionally, the correct frametype has to be included [here](https://github.com/PX4/Firmware/blob/f885d2274a563278b5ca501e4c19365ad0cd790d/src/modules/mavlink/mavlink_receiver.cpp#L1387) for the thrust setpoint.

### PR DESCRIPTION
**Pull request by ThomasRigi**
**Describe problem solved by the proposed pull request**
Partly fixes [#13092](https://github.com/PX4/Firmware/issues/13092) : 
Fixes offboard attitude control for VTOLs of type VTOL_TYPE_RESERVED2 in SITL.

Does not work with HIL and the big question is why?

**Test data / coverage**
Successfully tested in SITL with MAVSDK function `Offboard::set_attitude(Offboard::Attitude attitude)`: 
https://logs.px4.io/plot_app?log=870a9ef1-c346-4a18-beb9-1afb8c6a287a

Not working in HIL (with Pixhawk 4 as hardware) : 
https://logs.px4.io/plot_app?log=f714ce4f-828c-45bc-8753-f95fd6b2bbab

Not tested on real drones.

**Describe your preferred solution**
Changed the ORB ID of the attitude setpoint publication for VTOL_TYPE_RESERVED2. Other VTOL_TYPE_* vehicles can easily be added in the switch. Downside: needs a new ORB Publication with the new API.

**Describe possible alternatives**
Could also use vehicle_status.is_vtol flag , but that would need another ORB subscription to access that information.
Also, with the old ORB API, it would be possible to only have one publication in the header file and set the ORB ID accordingly in the switch before publication

**Additional context**
I could not figure out why it doesn't work in HIL, that's a big point of confusion for me. Would be good if someone else could have a look at that. :)

### CODE REVIEW

**Comment by julianoes** [comment]
I'm confused. What's `VTOL_TYPE_RESERVED2`?

**Comment by ThomasRigi** [comment]
@julianoes All standard VTOL frames I have looked at use this MAV_TYPE, so I thought they were equivalent. Maybe @sfuhrer or @RomanBapst can confirm?

**Comment by sfuhrer** [comment]
> @julianoes All standard VTOL frames I have looked at use this MAV_TYPE, so I thought they were equivalent. Maybe @sfuhrer or @RomanBapst can confirm?

Looks like you're correct. VTOL_TYPE_RESERVED2 is MAV_TYPE 22, which we fly for standard VTOL. No clue why it's not MAV_TYPE_VTOL_QUADROTOR (20) actually... @RomanBapst ?

https://github.com/mavlink/c_library_v2/blob/31cc5bb39e1cb69f5e608e8e2cac8a4d57332e1c/common/common.h#L91-L95

**Comment by ThomasRigi** [comment]
MAV_TYPE_VTOL_QUADROTOR (20) has a "tailsitter" comment... (doesn't make too much sense to me though)

**Comment by julianoes** [comment]
Aha, I see.

**Review by julianoes — COMMENTED** [review]
This looks correct but please run `make format`.

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -1397,6 +1397,7 @@ MavlinkReceiver::handle_message_set_attitude_target(mavlink_message_t *msg)
 						case MAV_TYPE_OCTOROTOR:
 						case MAV_TYPE_TRICOPTER:
 						case MAV_TYPE_HELICOPTER:
+						case MAV_TYPE_VTOL_RESERVED2:
```
Comment (Jaeyoung-Lim) [inline src/modules/mavlink/mavlink_receiver.cpp]: Is this specific only for this VTOL type?

Comment (ThomasRigi) [reply src/modules/mavlink/mavlink_receiver.cpp]: I'm not sure how thrust is handled in the other VTOL types, so I preferred to leave it blank for those types. I only work on quadplanes.

Also, my newly added `switch` on lines 1410-1417 eventually needs to be updated with the other VTOL types (preferably by someone who knows the other types, not that I make a stupid mistake)

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 2]
```diff
@@ -1397,6 +1397,7 @@ MavlinkReceiver::handle_message_set_attitude_target(mavlink_message_t *msg)
...
 							att_sp.thrust_body[2] = -set_attitude_target.thrust;
```
Comment (jlecoeur) [inline src/modules/mavlink/mavlink_receiver.cpp]: This will work only in MC mode, `att_sp.thrust_body[0]` should be used when in fixed mode.

**Review by jlecoeur — COMMENTED** [review]
@ThomasRigi (and @irsdkv in #13140), you both report that you tested your PRs with MAVSDK, however the two solutions are not  the same. Is it really needed to publish on topic `mc_virtual_attitude_setpoint`?

**Comment by irsdkv** [comment]
> Is it really needed to publish on topic `mc_virtual_attitude_setpoint`?

In case of mc mode we should use this topic for VTOL:
https://github.com/PX4/Firmware/blob/f885d2274a563278b5ca501e4c19365ad0cd790d/src/modules/vtol_att_control/vtol_att_control_main.cpp#L355

File: src/modules/mavlink/mavlink_receiver.cpp (hunk 2 above)
Comment (irsdkv) [reply src/modules/mavlink/mavlink_receiver.cpp]: Tried to check that:
https://github.com/PX4/Firmware/pull/13141/commits/db987e90f6134ec38d524104c3923f83312fad1e

(Due to [this](https://github.com/PX4/Firmware/blob/f885d2274a563278b5ca501e4c19365ad0cd790d/src/modules/vtol_att_control/vtol_att_control_main.cpp#L387) `switch`)

Comment (irsdkv) [reply src/modules/mavlink/mavlink_receiver.cpp]: Now checked and filling in true `thrust_body` field : https://github.com/PX4/Firmware/pull/13141/files

**Comment by ThomasRigi** [comment]
As my PR is much more of a stub than @irsdkv's, I will close this one.

**Pull request closed without merging**